### PR TITLE
PouchLink improvements

### PR DESCRIPTION
**Pull request by y-lohse**
PouchLink can now be configured to replicate certain doctypes only in one direction, which is useful for `io.cozy.files` for example that can't be replicated to the stack, but we'd still like to have the content in the local database. 

The API is debatable; fo the main use case (bi-directional replication), I think the current API is great (`new PouchLink({
  doctypes: ['io.cozy.whatever'] })`) and doesn't need to move. I added support for a second, more advanced format, which is `new PouchLink({
  doctypes: { readWrite: ['io.cozy.whatever'], readOnly: ['io.cozy.files'] } })`. It's flexible, duplication free and extandable with a `writeOnly` if we ever need that.

This PR also adds detection for next pages when results are paginated with `skip` and `limit`, which is how the `pouch.find` API uses.

### CODE REVIEW

**Comment by Crash--** [comment]
which is useful for io.cozy.doctypes > io.cozy.files

File: packages/cozy-pouch-link/src/PouchManager.js [hunk 1]
```diff
@@ -31,14 +31,21 @@ const humanTimeDelta = timeMs => {
 }
 
 /* Create a cancellable promise for replication with default options */
-const startReplication = (pouch, getReplicationURL) => {
+const startReplication = (
```
Comment (Crash--) [inline packages/cozy-pouch-link/src/PouchManager.js:34]: can we have a docstring here ?

File: packages/cozy-pouch-link/src/PouchManager.js [hunk 2]
```diff
@@ -31,14 +31,21 @@ const humanTimeDelta = timeMs => {
...
+  pouch,
+  replicateFromRemoteOnly,
```
Comment (Crash--) [inline packages/cozy-pouch-link/src/PouchManager.js:36]: I'm not a big fan of rewording the concept. One time we use : readOnly, and other ReplicateFromRemoteOnly. Can't we use one for both?

File: packages/cozy-pouch-link/src/PouchManager.spec.js [hunk 3]
```diff
@@ -69,13 +69,28 @@ describe('PouchManager', () => {
...
 
-  it('should periodically call replicate', async () => {
+  it('should periodically call sync', async () => {
     manager.startReplicationLoop()
     await sleep(1000)
     const pouch = manager.getPouch('io.cozy.todos')
     expect(pouch.sync.mock.calls.length).toBeGreaterThan(5)
   })
 
+  it('should only replicate in one direction for some doctypes', async () => {
```
Comment (Crash--) [inline packages/cozy-pouch-link/src/PouchManager.spec.js:79]:
```suggestion
  it('should only replicate from the remote for some doctypes', async () => {
```

File: packages/cozy-pouch-link/src/PouchManager.js (hunk 2 above)
Comment (y-lohse) [reply packages/cozy-pouch-link/src/PouchManager.js:36]: Fair, i've renamed the variables to `ReplicateFromRemoteOnly`. However for the public API i'd rather keep `readOnly`/`readWrite` since I think `ReplicateFromRemoteOnly`/Replicate...BothWays?`is too wordy and the distinction between bidirectional and the others is hard to express.

Comment (Crash--) [reply packages/cozy-pouch-link/src/PouchManager.js:44]: 👍

File: packages/cozy-pouch-link/src/PouchManager.js [hunk 4]
```diff
@@ -30,15 +30,30 @@ const humanTimeDelta = timeMs => {
...
 
-/* Create a cancellable promise for replication with default options */
-const startReplication = (pouch, getReplicationURL) => {
+/**
+ * startReplication - Create a cancellable promise for replication with default options
+ *
+ * @param {object} pouch                 Pouch database instance
+ * @param {boolean} replicateFromRemoteOnly Whether the replication should come from the remote only or be bidirectional
+ * @param {Function} getReplicationURL A function that should return the remote replication URL
+ *
```
Comment (ptbrowne) [inline packages/cozy-pouch-link/src/PouchManager.js:39]: I think we should add @private here.

File: packages/cozy-pouch-link/src/PouchManager.js [hunk 5]
```diff
@@ -30,15 +30,30 @@ const humanTimeDelta = timeMs => {
...
   const promise = new Promise((resolve, reject) => {
     const url = getReplicationURL()
-    replication = pouch.sync(url, {
-      batch_size: 1000 // we have mostly small documents
-    })
+    replication = replicateFromRemoteOnly
+      ? pouch.replicate.from(url)
+      : pouch.sync(url, {
+          batch_size: 1000 // we have mostly small documents
+        })
```
Comment (ptbrowne) [inline packages/cozy-pouch-link/src/PouchManager.js:55]: batchSize options should be passed to replicate.from .

**Comment by y-lohse** [comment]
After a chat with @ptbrowne I reworked the PR a little bit:

- The API has changed and I added a blip about it in the documentation
- CouchLink doesn't touch the replication options at all, it passes them to CouchManager
- CouchManager now supports all 3 types of replication, as well as receiving any other replication option

File: packages/cozy-pouch-link/src/CozyPouchLink.js [hunk 6]
```diff
@@ -49,17 +49,29 @@ const normalizeAll = (docs, doctype) => {
  * to respond to queries and mutations.
  */
 class PouchLink extends CozyLink {
+  /**
+   * constructor - Initializes a new PouchLink
+   *
+   * @param {object} [opts={}]
+   * @param {number} [opts.replicationInterval] Milliseconds between replications
+   * @param {(object)} [opts.doctypes] List of doctypes to replicate
```
Comment (ptbrowne) [inline packages/cozy-pouch-link/src/CozyPouchLink.js:57]:
```suggestion
   * @param {Array<string>} [opts.doctypes] List of doctypes to replicate
```

File: packages/cozy-pouch-link/src/CozyPouchLink.js [hunk 7]
```diff
@@ -49,17 +49,29 @@ const normalizeAll = (docs, doctype) => {
...
+   * @param {(object)} [opts.doctypesReplicationOptions] A dictionnary of doctypes and associated configuration options. All pouch replication options can be used, as well as the "strategy" option that determines which way the replication is done (can be "sync", "fromRemote" or "toRemote")
```
Comment (ptbrowne) [inline packages/cozy-pouch-link/src/CozyPouchLink.js:58]:
```suggestion
   * @param {object[]} opts.doctypesReplicationOptions A mapping from doctypes to replication options. All pouch replication options can be used, as well as the "strategy" option that determines which way the replication is done (can be "sync", "fromRemote" or "toRemote")
```
